benchmarks/domains/Hardness.py

```diff
@@ -0,0 +1,266 @@
+            objective=objective,
+        ),
+    }
+
+    return simulate_scenarios(
+        scenarios,
+        df_lookup_target,
+        batch_size=settings.batch_size,
+        n_doe_iterations=settings.n_doe_iterations,
+        n_mc_iterations=settings.n_mc_iterations,
+        impute_mode="error",
+    )
+
+
+def hardness_transfer_learning(settings: ConvergenceBenchmarkSettings) -> DataFrame:
+    """Integrated hardness benchmark, transfer learning with different initialized data sizes
+
+    Inputs:
+        B   discrete    {0.8, 0.66666667, 0.92307692 ...}   |B| = 13
+        Sc  discrete    {0.,  0.00384615, 0.01923077 ...}   |Sc| = 26
+        Cr  discrete    {0.01, 0.06, 0.1 ...}               |Cr| = 20
+        Y   discrete    {0., 0.07307692, 0.05769231 ...}    |Y| = 31
+        Zr  discrete    {0., 0.07307692, 0.05769231 ...}    |Zr| = 19
+        Gd  discrete    {0., 0.03968254, 0.01587302 ...}    |Gd| = 12
+        Hf  discrete    {0., 0.008, 0.02 ...}               |Hf| = 13
+        Ta  discrete    {0., 0.006, 0.008 ...}              |Ta| = 17
+        W   discrete    {0.19, 0.14, 0.1 ...}               |W| = 30
+        Re  discrete    {0., 0.2, 0.33333 ...}              |Re| = 15
+    Output: discrete
+    Objective: maximization
+    """
+    parameters = []
+
+    # For each column in df_searchspace except the last one, create a NumericalDiscreteParameter
+    for column in df_searchspace.columns[:-1]:
+        discrete_parameter_i = NumericalDiscreteParameter(
+            name=column,
+            values=np.unique(df_searchspace[column]),
+            tolerance=0.0,
+        )
+        parameters.append(discrete_parameter_i)
+
+    task_parameter = TaskParameter(
+        name="Function",
+        values=["targetFunction", "sourceFunction"],
+        active_values=["targetFunction"],
+    )
+    parameters.append(task_parameter)
+
+    objective = NumericalTarget(name="Target", mode="MAX").to_objective()
+
+    searchspace = SearchSpace.from_dataframe(df_searchspace, parameters=parameters)
+
+    # Seperate campaign for different initial data size, preventing unintended data overwriting
+    scenarios: dict[str, Campaign] = {
+        f"{n} Initial Data": Campaign(searchspace=searchspace, objective=objective)
+        for n in (2, 4, 6, 30)
+    }
+
+    # Create an iterable of datasets with different initial sizes
+    initial_data_sets = [df_lookup_source.sample(n) for n in (2, 4, 6, 30)]
+
+    return simulate_scenarios(
+        scenarios,
+        df_lookup_target,
+        initial_data=initial_data_sets,
+        batch_size=settings.batch_size,
+        n_doe_iterations=settings.n_doe_iterations,
+        impute_mode="error",
+    )
+
+
+benchmark_config = ConvergenceBenchmarkSettings(
+    batch_size=1,
+    n_doe_iterations=20,
+    n_mc_iterations=5,
+)
+
+hardness_benchmark = ConvergenceBenchmark(
+    function=hardness,
+    settings=benchmark_config,
+    optimal_target_values=None,

```

Comment:
How come we don't know the optimal value? This should be clear from the dataset, no?